src/tmpfiles/tmpfiles.c

```diff
@@ -1274,12 +1274,11 @@ static int create_file(Item *i, const char *path) {
 
         assert(i);
         assert(path);
-        assert(IN_SET(i->type, CREATE_FILE, TRUNCATE_FILE));
+        assert(i->type == CREATE_FILE);
 
-        /* FIXME: O_TRUNC is unspecified if file is neither a regular file nor a
-         * fifo nor a terminal device. Therefore we should fail if file is
-         * anything but a regular file with 'F'. */
-        flags = O_CREAT|O_NOFOLLOW|(i->type == CREATE_FILE ? O_EXCL : O_TRUNC);
+        /* 'f' operates on regular files exclusively. */
+
+        flags = O_CREAT|O_EXCL|O_NOFOLLOW;
```

Comment:
`flags` has just a single use. I'd prefer to not have the variable and just put the flags definition in `open` below, and add the comment there.